tests/unit/pluginhandler/test_pluginhandler.py

```diff
@@ -781,6 +786,143 @@ def test_makedirs_with_existing_dirs(self):
             self.assertTrue(os.path.exists(d), '{} does not exist'.format(d))
 
 
+class NextLastStepTestCase(unit.TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.handler = self.load_part('test_part')
+
+    def test_pull(self):
+        self.assertThat(self.handler.last_step(), Equals(None))
+        self.assertThat(self.handler.next_step(), Equals('pull'))
+
+        self.handler.pull()
+
+        self.assertThat(self.handler.last_step(), Equals('pull'))
+        self.assertThat(self.handler.next_step(), Equals('build'))
+
+    def test_build(self):
+        self.assertThat(self.handler.last_step(), Equals(None))
+        self.assertThat(self.handler.next_step(), Equals('pull'))
+
+        self.handler.build()
+
+        self.assertThat(self.handler.last_step(), Equals('build'))
+        self.assertThat(self.handler.next_step(), Equals('stage'))
+
+    def test_stage(self):
+        self.assertThat(self.handler.last_step(), Equals(None))
+        self.assertThat(self.handler.next_step(), Equals('pull'))
+
+        self.handler.stage()
+
+        self.assertThat(self.handler.last_step(), Equals('stage'))
+        self.assertThat(self.handler.next_step(), Equals('prime'))
+
+    def test_prime(self):
+        self.assertThat(self.handler.last_step(), Equals(None))
+        self.assertThat(self.handler.next_step(), Equals('pull'))
+
+        self.handler.prime()
+
+        self.assertThat(self.handler.last_step(), Equals('prime'))
```

Comment:
seeing these together I think we should call this `previous_step`

Reply:
Yeah fine by me, although this is pretty old and used in a number of places. Maybe a followup?

Reply:
follow up is fine, it is all internal